share/janitor/tasks.py

```diff
@@ -2,19 +2,40 @@
 
 import celery
 
+from django.db.models import Exists
+from django.db.models import OuterRef
+
 from share import tasks
 from share.models import RawDatum
+from share.models import NormalizedData
 
 
 logger = logging.getLogger(__name__)
 
 
 @celery.shared_task(bind=True)
-def rawdata_janitor(self, limit=100):
+def rawdata_janitor(self, limit=500):
     """Find RawDatum that do not have a NormalizedData process them
     """
     count = 0
-    for rd in RawDatum.objects.select_related('suid__source_config').filter(normalizeddata__isnull=True).order_by('id')[:limit].iterator():
+
+    # NOTE: Do NOT use .iterator here. It will create a temporary table and eat disk space like no other
```

Comment:
Was `.iterator` the problem, or was it the join from `.filter(normalizeddata__isnull=True)`? I'm trying to better understand nuances of Django and SQL, and Django [claims that](https://docs.djangoproject.com/en/1.11/ref/models/querysets/#iterator)
> internally, the default iterator calls iterator() and caches the return value

but I wouldn't be shocked if something weirder was going on.